### DISCUSSION TITLE
A fatal error occurred: Corrupt data, expected 0x1000 bytes but received 0xfff bytes

### DISCUSSION THREAD

**Discussion by eigenphase** [emporia-vue-local/esphome#133] [Q&A]
I followed the instructions on https://github.com/emporia-vue-local/esphome precisely but I get the following error:
```

 esptool.py -b 115200 read_flash 0 0x800000 flash_contents.bin
esptool.py v3.3.1
Found 4 serial ports
Serial port /dev/ttyUSB0
Connecting....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting...
Detecting chip type... ESP32
Chip is ESP32-D0WD (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: a8:48:fa:98:81:2c
Uploading stub...
Running stub...
Stub running...
4096 (0 %)
A fatal error occurred: Corrupt data, expected 0x1000 bytes but received 0xfff bytes
```

What could be the issue?
One thing I could think of: Do the data voltages need to be 3.3V or 5V?
My serial adapter has two settings: 3.3V and 5.0V. While I connect the 5V as a supply, as instructed, I still leave the jumper for the DATA voltages at 3.3V. I do not want to fry my new Vue 2.

Should they be 5V, maybe?

**Comment by flaviut** [comment]
> Should they be 5V, maybe?

No. Data needs to be 3.3V.

I haven't seen this before. Maybe https://github.com/emporia-vue-local/esphome/discussions/31 is related to your issue.

But it really looks to me like there's a connection that's only 99% reliable. Try using new jumper wires.

**Reply by eigenphase** [reply]
Thank you!
Actually, what helped is to use the 3.3V supply instead of the 5V supply!

And good I didn’t try the 5V data voltage :-)

As a suggestion, would be great to explicitly add this (3.3V data) to the instructions such that it’s unambiguous.
